appscale/tools/appscale_stats.py

```diff
@@ -0,0 +1,657 @@
+
+
+def get_marked(data, mark):
+  """
+  Marks data in a specified mark.
+
+  Args:
+    data: An object to be marked in.
+    mark: A string representing one of the existing marks
+      ('red', 'green' or 'bold').
+
+  Returns:
+    A string marked in.
+  """
+  marks = {
+    "red": "\033[91m",
+    "green": "\033[92m",
+    "bold": "\033[1m"
+  }
+  return marks[mark] + str(data) + "\033[0m"
+
+
+def render_loadavg(loadavg):
+  """
+  Renders loadavg information.
+
+  Args:
+    loadavg: A dict representing loadavg values.
+
+  Returns:
+    A string with information about node loadavg
+    last 1, 5 and 15 minuts in format: "last 1min / last 5min / last 15min"
+    and marked in red if loadavg is more than 2.0.
+  """
+  limit_value = 2.0
+  last_1 = loadavg["last_1min"]
+  last_5 = loadavg["last_5min"]
+  last_15 = loadavg["last_15min"]
+
+  return "{} / {} / {}".format(
+    last_1 if last_1 < limit_value else get_marked(last_1, "red"),
+    last_5 if last_5 < limit_value else get_marked(last_5, "red"),
+    last_15 if last_15 < limit_value else get_marked(last_15, "red")
+  )
+
+
+def render_partitions(partitions, verbose):
+  """
+  Renders partitions information.
+
+  Args:
+    partitions: A dict representing partition values.
+    verbose: A boolean - render all partitions if True,
+      add only three the most used partitions if False.
+
+  Returns:
+    A string with information about partition values
+      marked partition and value in red if value > 90%.
+  """
+  part_list = [[part, 100 * value["used"] / value["total"]]
+               for part, value in
+               partitions.iteritems()]  # calculate percents
+
+  # sort from more used to less used
+  part_list.sort(key=lambda p: p[1], reverse=True)
+
+  partitions_info = [
+    "'{part}': {usage}%".format(part=part[0], usage=part[1])
+    if part[1] < 90
+    else get_marked(
+      "'{part}': {usage}%".format(part=part[0], usage=part[1]), "red"
+    )
+    for part in part_list
+  ]
+
+  if not verbose and len(partitions_info) > 3:
+    partitions_info = partitions_info[:3] + ["..."]
+
+  return ", ".join(partitions_info)
+
+
+def render_memory(memory):
+  """
+  Renders memory information.
+
+  Args:
+    memory: A dict representing memory values.
+
+  Returns:
+    A string with information about node memory
+    in format: "available % / available MB".
+  """
+  return "{}% ({} MB)".format(
+    (100 * memory["available"] / memory["total"]),
+    (memory["available"] / 1024 / 1024)
+  )
+
+
+def sort_process_stats(process_stats, column, top, reverse=True):
+  """
+  Sorts process statistics by specified column.
+
+  Args:
+    process_stats: A list of process statistics.
+    column: An int representing a column number the list should be sorted by.
+    top: An int representing a process count to be printed.
+    reverse: A boolean to reverse or not reverse sorted list.
+
+  Returns:
+    A list of top processes sorted by specified column.
+  """
+  if top <= 0:
```

Comment:
I would avoid using possible value as an exceptional case. I don't think a user would ever specify `--top 0`, but I also don't think it would be intuitive for the tools to print everything if that value _was_ explicitly specified.

In this case, it would be conventional to just use `None` to signify that the user did not specify an argument by just removing the `default` parameter on this argument.